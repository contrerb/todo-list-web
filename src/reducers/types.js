export default {
  STATES_GET_REQUEST: "STATES_GET_REQUEST",
  STATES_GET_RECEIVE: "STATES_GET_RECEIVE",
  STATES_GET_ERROR: "STATES_GET_ERROR",
  STATES_CLEAR: "STATES_CLEAR",
  STATES_CLEAR_ERROR: "STATES_CLEAR_ERROR",
  I18N_LANGUAGE: "I18N_LANGUAGE",
  I18N_LOCALE: "I18N_LOCALE"
};
