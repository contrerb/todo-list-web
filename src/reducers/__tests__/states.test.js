import reducer, { actions } from "../states";
import types from "../types";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { apiMiddleware } from "@cdkglobal/react-core-reduxapi";

const middlewares = [thunk, apiMiddleware];
const mockStore = configureMockStore(middlewares);

it("returns the initial state", () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

it("handles STATES_GET_REQUEST", () => {
  expect(
    reducer(undefined, { type: types.STATES_GET_REQUEST })
  ).toMatchSnapshot();
});

it("handles STATES_GET_RECEIVE", () => {
  expect(
    reducer(undefined, {
      type: types.STATES_GET_RECEIVE,
      response: ["a", "b", "c"]
    })
  ).toMatchSnapshot();
});

it("handles STATES_GET_ERROR", () => {
  expect(
    reducer(undefined, {
      type: types.STATES_GET_ERROR,
      apierror: { message: "error" }
    })
  ).toMatchSnapshot();
});

it("handles STATES_CLEAR_ERROR", () => {
  expect(
    reducer(
      { list: [], status: { isFetching: false, error: "error" } },
      actions.clearError()
    )
  ).toMatchSnapshot();
});

it("handles STATES_CLEAR", () => {
  expect(reducer({ list: ["a", "b", "c"] }, actions.clear())).toMatchSnapshot();
});

it("fetches states", async () => {
  fetch.mockResponse(JSON.stringify(["a", "b", "c"]));
  const store = mockStore({});
  await store.dispatch(actions.fetch());
  expect(store.getActions()).toMatchSnapshot();
});

it("handles fetch errors", async () => {
  fetch.mockReject(new Error("An error occurred"));
  const store = mockStore({});
  await store.dispatch(actions.fetch());
  expect(store.getActions()).toMatchSnapshot();
});
