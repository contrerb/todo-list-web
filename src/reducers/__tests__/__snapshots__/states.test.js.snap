// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches states 1`] = `
Array [
  Object {
    "type": "STATES_GET_REQUEST",
  },
  Object {
    "response": "[\\"a\\",\\"b\\",\\"c\\"]",
    "type": "STATES_GET_RECEIVE",
  },
]
`;

exports[`handles STATES_CLEAR 1`] = `
Object {
  "list": Array [],
  "status": Object {
    "error": "",
    "isFetching": false,
  },
}
`;

exports[`handles STATES_CLEAR_ERROR 1`] = `
Object {
  "list": Array [],
  "status": Object {
    "error": "",
    "isFetching": false,
  },
}
`;

exports[`handles STATES_GET_ERROR 1`] = `
Object {
  "list": Array [],
  "status": Object {
    "error": "error",
    "isFetching": false,
  },
}
`;

exports[`handles STATES_GET_RECEIVE 1`] = `
Object {
  "list": Array [
    "a",
    "b",
    "c",
  ],
  "status": Object {
    "error": "",
    "isFetching": false,
  },
}
`;

exports[`handles STATES_GET_REQUEST 1`] = `
Object {
  "list": Array [],
  "status": Object {
    "error": "",
    "isFetching": true,
  },
}
`;

exports[`handles fetch errors 1`] = `
Array [
  Object {
    "type": "STATES_GET_REQUEST",
  },
  Object {
    "apierror": [Error: An error occurred],
    "type": "STATES_GET_ERROR",
  },
]
`;

exports[`returns the initial state 1`] = `
Object {
  "list": Array [],
  "status": Object {
    "error": "",
    "isFetching": false,
  },
}
`;
